library/init/data/nat/lemmas.lean

```diff
@@ -730,5 +770,148 @@ lemma pred_inj : ∀ {a b : nat}, a > 0 → b > 0 → nat.pred a = nat.pred b 
 
 /- TODO(Leo): sub + inequalities -/
 
+/- div -/
+
+theorem zero_div : ∀ m : ℕ, 0 / m = 0
+  | zero := by refl
+  | (succ n) := by refl
+
+theorem div_le_div
+: ∀ {y x}, x ≤ y → ∀ z : ℕ, x / z ≤ y / z :=
```

Comment:
Actually, I don't think that stronger theorem holds: consider x = 3, y = 2 and z = 2.

x / z = 1
y / z = 1
but 3 ≤ 2 does not hold.

I could prove a Galois connection between / and * though:

x / z ≤ y ↔ x ≤ y * z

Together with indirect equality (see below) it could provide a basis for simpler proofs about division.
theorem indirect_equality [weak_order α] (x y : α) : x = y ↔ (∀ z : α, x ≤ z ↔ y ≤ z)

Should I go ahead?